draft-ietf-mls-protocol.md

```diff
@@ -5067,12 +4672,12 @@ all the members of the group that will process the Commit.
 
 ## Credential Extensibility
 
-In order to ensure that MLS provides meaningful authentication it is important
+In order to ensure that MLS provides meaningful authentication, it is important
 that each member is able to authenticate some identity information for each
 other member.  Identity information is encoded in Credentials, so this property
 is provided by ensuring that members use compatible credential types.
 
-The types of credential that may be used in a group is restricted to what all
+The only types of credential that may be used in a group are those that all
 members of the group support, as specified by the `capabilities` field of each
 LeafNode in the ratchet tree. An application can introduce new credential types
```

Comment:
This is an unnecessary imposition and makes it impossible to add support for new credential types for some members of a group, and makes KeyPackage more complicated. If a specific credential type is required by all the members of the group, it should be included in `required_capabilities`.  The specific use case of unknown credentials was discussed in various interim meetings and was a major motivator for moving the signature out of the credential and into the LeafNode.

```suggestion
If a specific credential type is included in the `required_capabilities` extension
in the GroupContext, then all members of the group MUST support that credential
type, as specified by the `capabilities` field of each LeafNode in the ratchet tree.
Credential types that are unknown to a member and not included the
`required_capabilities` extension are ignored by the member. This allows a subset
of members to introduce additional functionality without requiring universal
support for a new credential type. An application can introduce new credential types
```

Reply:
I remember the discussions a bit differently, namely that we included the signature key in the leaf node so that we can have easier extensibility for credentials, but *not* that we can now have arbitrary combinations of credentials. I'm worried that this weakens authentication guarantees when clients can only verify a signature but no longer correlate it to an identity. It's also not clear to me why we can't do what we do with other extensions: first make sure every client in the group supports it as advertized in its capabilities and *only then* make it a required capability before using it.

Reply:
The part I'm worried about here is `Credential types that are unknown to a member and not included the
required_capabilities extension are ignored by the member.` This seems to conflict with the fact that every credential ` MUST be validated with the AS`, since a member can't verify a credential it doesn't support? Note that required capabilities are optional, so credentials must be verified even without them.

Reply:
I second @raphaelrobert and @mulmarta. Allowing credentials that are not supported by a subset of group members and which they can thus only ignore seems like a very bad idea from an authentication standpoint. @rohan-wire can you maybe elaborate what kind of use-case you had in mind? Because I'm having trouble thinking of one. It would have to be a situation, where different applications are interoperating and where clients are okay with group members that they cannot authenticate at all. For such a group, one might was well use basic credentials.

Reply:
You provide an example in which you claim there's no downside to your proposed changes. Here's a counter-example: Suppose a group is to use X.509 and X.509 only. With the proposed changes, a subset of the clients could start using BasicCredentials and the remainder of the clients would have to comply with that, since BasicCredentials would simply be an unknown credential to them.
We would need better reasoning about the relaxation, however I'm doubtful that it's a good idea to do such a potentially impactful change during AUTH48, without the possibility to fully consider all consequences.

Reply:
Why wouldn't those clients just use `required_capabilities` if they only want to use X.509?
I also find it a bit preposterous the idea that an implementation would not understand what `basic` is, as opposed to administratively refusing to use it.

I am sympathetic about the AUTH48 comment, but that would indicate this is better fixed as an errata than an extension.

Reply:
Without getting into the details of the discussion here -- this is a significant change for this late.  I would have been skeptical but open earlier in the process; at this point, it's clearly too big.  And it doesn't seem like anyone is in support besides @rohan-wire.  I'm going to decline to include this change unless the chairs tell me otherwise.

FWIW, I would also argue that this change should not be accepted as an erratum.  It is not an error in the spec, it is a deliberate decision by the working group.

That said, you could definitely write a small, no-content `allow_unknown_credentials` GroupContext extension that just says, "In this group, it's OK to use credentials that not everyone supports".  Then you just switch on the presence of that extension to turn off the credential support checking.  You still have an everyone-has-to-upgrade problem, but only with regard to that extension, not every credential type.